R/auditCommonFunctions.R

```diff
@@ -2040,104 +2120,47 @@
+                                  prior = prior,
+                                  increase = options[["sampleSizeIncrease"]])
+          n[i] <- result[["sampleSize"]]
+        }
+      })
       
-      if(options[["bayesianAnalysis"]]){
+      if(class(rightPlotError) != "try-error"){
         
-        # Create a prior distribution that incorporates the existing information
-        prior <- jfa::auditPrior(materiality = performanceMateriality,
-                                 confidence = confidence,
-                                 expectedError = parentOptions[["expectedErrors"]],
-                                 likelihood = parentOptions[["likelihood"]],
-                                 N = N,
-                                 ir = ir,
-                                 cr = cr,
-                                 method = options[["priorConstructionMethod"]],
-                                 sampleN = options[["sampleN"]],
-                                 sampleK = options[["sampleK"]],
-                                 factor = options[["factor"]],
-                                 pHplus = 1 - options[["pHmin"]],
-                                 pHmin = options[["pHmin"]])
+        dPlot <- data.frame(x = c("0", "1", "2", "3"), y = c(2, 2, 2, 2))
+        dPlot$x <- factor(dPlot$x, levels = c("3", "2", "1", "0"))
+        
+        plot <- ggplot2::ggplot(data = dPlot, mapping = ggplot2::aes(x = x, y = y, fill = x)) +
+          ggplot2::geom_bar(stat = "identity", col = "black", size = 1) +
+          ggplot2::scale_y_continuous(name = "", breaks = NULL, limits = c(0, 2.3)) +
+          ggplot2::scale_x_discrete(name = "") +
+          ggplot2::scale_fill_manual(values = c("red", "darkorange1", "orange", "#7FE58B")) +
+          ggplot2::coord_flip() +
+          ggplot2::labs(fill = "") +
+          ggplot2::annotate("text", y = c(0.5, 0.5, 0.5, 0.5), x = c(4, 3, 2, 1),
+                            label = paste0("n = ", n), size = 6, vjust = 0.5, hjust = 0.2, fontface = "italic")
+        plot <- jaspGraphs::themeJasp(plot, sides = "", legend.position = "none") + jfaTheme
+        
+        figure$plotObject <- plot
       } else {
-        prior <- FALSE
-      }
-      
-      n <- numeric(length(1:4))
-      for(i in 1:4){
-        result <- jfa::planning(materiality = performanceMateriality,
-                                confidence = confidence,
-                                likelihood = parentOptions[["likelihood"]],
-                                expectedError = i - 1,
-                                N = N,
-                                minPrecision = minPrecision,
-                                prior = prior,
-                                increase = options[["sampleSizeIncrease"]])
-        n[i] <- result[["sampleSize"]]
+        if(jaspBase:::.extractErrorMessage(rightPlotError) == "Sample size could not be calculated, you may want to increase the maxSize argument."){
```

Comment:
should this be `gettext`ed?

Reply:
I don't think so, this is an error message returned by the `jfa` package.